var/spack/repos/builtin/packages/asio/package.py

```diff
@@ -62,6 +62,7 @@ class Asio(AutotoolsPackage):
     version("1.16.1", sha256="e40bbd531530f08318b7c7d7e84e457176d8eae6f5ad2e3714dc27b9131ecd35")
     version("1.16.0", sha256="c87410ea62de6245aa239b9ed2057edf01d7f66acc3f5e50add9a29343c87512")
 
+    depends_on("c", type="build")
```

Comment:
No idea why `asio` would need a C compiler. I am completely inexperienced concerning autoconf/automake so take that with a grain of salt but I didn't see any reference to C.

Reply:
Yeah, this doesn't seem to be necessary. The CI configurations for asio also only set `CXX`; there are only `.cpp` files, no `.c` files, in the repo; and there's no mention of `CFLAGS`.

Reply:
`configure` is probing for a C compiler. Without this I get:
![Screenshot from 2024-11-20 17-47-55](https://github.com/user-attachments/assets/44129c21-cae5-4120-b462-0d455b48c1b1)

In #45189